### PR TITLE
upgrade to sbt 1.0

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

Upgrades SBT to version 1.0.4

## What is the value of this and can you measure success?

Continued SBT support, faster builds

## Tested in CODE?

We will

### CODE REVIEW

File: project/ProjectSettings.scala [hunk 1]
```diff
@@ -25,7 +25,6 @@ object ProjectSettings {
     publishArtifact in (Compile, packageDoc) := false,
     sources in (Compile,doc) := Seq.empty,
     doc in Compile := target.map(_ / "none").value,
-    incOptions := incOptions.value.withNameHashing(true),
```
Comment (QuarpT) [inline project/ProjectSettings.scala]: This is on by default now

File: project/VersionInfo.scala [hunk 2]
```diff
@@ -6,7 +6,7 @@ import sbt._
 import Keys._
 import org.eclipse.jgit.storage.file.FileRepositoryBuilder
 
-object VersionInfo extends Plugin {
```
Comment (QuarpT) [inline project/VersionInfo.scala]: Plugin has been removed, we must now use AutoPlugin

File: project/VersionInfo.scala [hunk 3]
```diff
@@ -25,7 +25,7 @@ object VersionInfo extends Plugin {
     }
   }
 
-  override val settings = Seq(
```
Comment (QuarpT) [inline project/VersionInfo.scala]: settings has been removed, we must override projectSettings or buildSettings

File: project/plugins.sbt [hunk 4]
```diff
@@ -16,18 +16,16 @@ resolvers ++= Seq(
...
-addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.0")
+addSbtPlugin("com.timushev.sbt" % "sbt-updates" % "0.3.3")
 
-addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "0.9.7")
+addSbtPlugin("com.gu" % "sbt-riffraff-artifact" % "1.0.0")
 
-addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.8.5")
+addSbtPlugin("com.typesafe.sbt" % "sbt-git" % "0.9.3")
 
-addSbtPlugin("org.jetbrains.teamcity.plugins" % "sbt-teamcity-logger" % "0.3.0")
```
Comment (QuarpT) [inline project/plugins.sbt]: There is no teamcity-logger plugin support for SBT 1.0 yet

File: project/ProjectSettings.scala [hunk 5]
```diff
@@ -139,25 +137,8 @@ object ProjectSettings {
...
-   * Modified this so that settings from "nonAutoPlugins" (plugins using the older sbt plugin API)
-   * are applied before settings defined in build.scala
-   *
-   * Required for resetting the `testResultLogger` in `frontendTestSettings` above
-   *
-   * Default:
-   *   AddSettings.allDefaults: AddSettings = seq(autoPlugins, buildScalaFiles, userSettings, nonAutoPlugins, defaultSbtFiles)
-   */
-  lazy val settingSetsOrder = {
-    import AddSettings._
```
Comment (QuarpT) [inline project/ProjectSettings.scala]: AddSettings has been dropped
https://github.com/sbt/sbt/issues/2617

File: project/plugins.sbt (hunk 4 above)
Comment (QuarpT) [reply project/plugins.sbt]: Looks like we need to enable https://confluence.jetbrains.com/display/TW/SBT+Runner+Plugin on TeamCity

File: project/project/Plugins.sbt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import sbt._
+
+// NOTE - NO source dependency plugins allowed in this project
+// either use a properly packaged dependency, or copy 'n paste the code
+// into the project.
+lazy val plugins = sbt.Project("build", file("."))
```
Comment (nicl) [inline project/project/Plugins.sbt]: Perhaps kill? Looks like not used.

**Review by nicl — APPROVED** [review]
Let's work out the TC change but looks good and will hopefully speed the build/compile times :)

**Pull request merged**